recipes/source.rb

```diff
@@ -3,6 +3,7 @@
 git '/usr/src/exabgp' do
   repository 'https://github.com/Exa-Networks/exabgp.git'
   reference node['exabgp']['source_version']
+  checkout_branch node['exabgp']['source_version']
```

Comment:
This is not necessary but exabgp checks for current version like this:
```
if [ $? -eq  0 ] && [ $GIT_TAG -eq 0 ]; then
   GIT_BRANCH=`git branch | grep "*" | awk '{ print $2}'`
   GIT_COMMIT=`git log -1 | head -1 | awk '{ print $2 }'`
   export EXABGP_VERSION="${GIT_BRANCH}-${GIT_COMMIT}"
fi
```

by default `git` resource checks out to `deploy` branch, and exabgp shows 
```
[root@source-42-centos-7 sbin]# ./exabgp -v
ExaBGP : deploy-0bcb7bb30f1f91cac168978b9eb0d1543799e00e
Python : 3.9.4 (default, Aug 17 2021, 07:09:31)  [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
Uname  : Linux source-42-centos-7.vagrantup.com 3.10.0-1127.19.1.el7.x86_64 #1 SMP Tue Aug 25 17:23:54 UTC 2020 x86_64
Root   : /usr/src/exabgp
```